homeassistant/components/mochad.py

```diff
@@ -64,15 +65,26 @@ def start_mochad(event):
 class MochadCtrl:
     """Mochad controller."""
 
-    def __init__(self, host, port):
-        """Initialize a PyMochad controller."""
-        super(MochadCtrl, self).__init__()
+    def __init__(self, hass, host, port):
+        """Initialize a PyMochad send-receive controller."""
         self._host = host
         self._port = port
 
-        from pymochad import controller
+        from pymochad_mqtt.controller import PyMochadMqtt
+
+        def mqtt_pub_callback(topic, payload, qos, retain):
+            """Call MQTT publish function."""
+            mqtt.publish(hass, topic, payload, qos, retain)
 
-        self.ctrl = controller.PyMochad(server=self._host, port=self._port)
+        self.ctrl_recv = PyMochadMqtt(mochad_server=self._host,
+                                      mochad_port=self._port,
+                                      mqtt_pub_callback=mqtt_pub_callback)
+        self.ctrl_recv.start()
+        _LOGGER.debug(
+            "PyMochadMqtt controller created for mochad %s:%s and mqtt",
+            host, port)
+        if self.ctrl_recv.connect_event.wait():
```

Comment:
Is this blocking? We're not allowed to block in the home assistant thread pool.

Reply:
this is inside of the setup method. Need to wait for the asynch init of the pymochad_mqtt. It's a short time and one time block (e.g. 1-2 sec).

Reply:
Instead of connecting when starting the other thread, break out that into a separate method, that we can call before starting the thread. Then we don't need to wait for the event.